### PR TITLE
share screen initiator flow was added

### PR DESCRIPTION
**Pull request by igormas**

### CODE REVIEW

File: dist/screenSharing/screenSharing.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    angular.module('znk.infra.screenSharing').service('ScreenSharingSrv',
+        ["UserProfileService", "InfraConfigSrv", "$q", "UtilitySrv", "ScreenSharingDataGetterSrv", "ScreenSharingStatusEnum", "ENV", "$log", function (UserProfileService, InfraConfigSrv, $q, UtilitySrv, ScreenSharingDataGetterSrv, ScreenSharingStatusEnum, ENV, $log) {
+            'ngInject';
+
+            var INITIATOR_ENUM = {
+                "VIEWER": 1,
+                "SHARER": 2
+            };
+
+            var isDashboardApp = ENV.appContext === 'dashboard';
```
Comment (assafshp) [inline dist/screenSharing/screenSharing.js]: ENV.appContext.tolowercase()

File: test/spec/screenSharing/screenSharingSrv.spec.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        _deps.testStorage.db.screenSharing[screenSharingDataGuid] = {
+            guid: screenSharingDataGuid,
+            sharerId: sharerId,
+            viewerId: viewerId,
+            status: _deps.ScreenSharingStatusEnum.PENDING_VIEWER.enum
+        };
+
+        _deps.ScreenSharingSrv.endSharing(screenSharingDataGuid);
+
+        var expectedRootScreenSharing = {};
```
Comment (assafshp) [inline test/spec/screenSharing/screenSharingSrv.spec.js]: The only diff between this test and the previous one is the status
 status: _deps.ScreenSharingStatusEnum.PENDING_VIEWER.enum
and the actio result - endsharing/confirmsharing maybe create a func for state change test ? not mandatory

File: test/spec/screenSharing/screenSharingSrv.spec.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        ' be update accordingly', function () {
+        var screenSharingDataGuid = 'guid';
+        spyOn(_deps.UtilitySrv.general, 'createGuid').and.returnValue(screenSharingDataGuid);
+
+        var sharerId = '11223344';
+        var viewerId = _deps.UserProfileService.__currUserId;
+
+        var sharerData = {
+            uid: sharerId,
+            isTeacher: false
```
Comment (assafshp) [inline test/spec/screenSharing/screenSharingSrv.spec.js]: do we have to have this property?

File: test/spec/screenSharing/screenSharingSrv.spec.js [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+            'ScreenSharingSrv',
+            'UserProfileService',
+            'testStorage',
+            'ScreenSharingStatusEnum',
+            'TestUtilitySrv',
+            'UtilitySrv',
+            'ENV'
+        ];
+
+        depsToInject.forEach(function (depName) {
```
Comment (assafshp) [inline test/spec/screenSharing/screenSharingSrv.spec.js]: I like it

File: test/mock/env/env.mock.js [hunk 5]
```diff
@@ -4,9 +4,12 @@
...
 
     angular.module('env.mock', [])
-        .service(mockENVServiceName, function(){
-             return {
-                'firebaseAppScopeName': 'test_app'
-             };
+        .service(mockENVServiceName, function () {
+            return {
+                'firebaseAppScopeName': 'test_app',
+                'appContext': 'student',
```
Comment (assafshp) [inline test/mock/env/env.mock.js]: don't we have too many params?

Comment (igormas) [inline test/mock/env/env.mock.js]: We must sync the dashboard and student ENV

File: dist/screenSharing/screenSharing.js (hunk 1 above)
Comment (igormas) [inline dist/screenSharing/screenSharing.js]: Fixed

File: test/spec/screenSharing/screenSharingSrv.spec.js (hunk 2 above)
Comment (igormas) [inline test/spec/screenSharing/screenSharingSrv.spec.js]: For simplicity lets leave it as is.

File: test/spec/screenSharing/screenSharingSrv.spec.js (hunk 3 above)
Comment (igormas) [inline test/spec/screenSharing/screenSharingSrv.spec.js]: Was change to isTeacher also in the service.

File: test/spec/screenSharing/screenSharingSrv.spec.js (hunk 4 above)
Comment (igormas) [inline test/spec/screenSharing/screenSharingSrv.spec.js]: 👍

**Pull request merged**